src/indexer/merger.rs

```diff
@@ -355,6 +377,55 @@ impl IndexMerger {
         }
     }
 
+    /// Checks if the readers are disjunct for their sort property and in the correct order to be
+    /// able to just stack them.
+    pub(crate) fn is_disjunct_and_sorted_on_sort_property(
+        &self,
+        sort_by_field: &IndexSortByField,
+    ) -> crate::Result<bool> {
+        let reader_and_field_accessors =
+            Self::get_reader_with_sort_field_accessor(&self.readers, sort_by_field)?;
+
+        let everything_is_in_order = reader_and_field_accessors
```

Comment:
Kudos on that. I love these variable used only to make the code more readable and provide the intent.